### PR TITLE
resolve #5417 ensure unlink order is correctly sorted

### LINKED ISSUE #5417: Respecting topological sorting for environment removal
Not sure if this is implemented in newer versions of `conda`, but it would be nice if environment removal performed a topological sort and uninstalled packages based on this sorting. Namely only removing packages that are not required for anything else first. For the most part this doesn't matter, but when `pre-unlink` scripts are involved it does.

### PR DESCRIPTION
**Pull request by kalefranz**
resolve #5417

### CODE REVIEW

**Comment by kalefranz** [comment]
Reviewer: Please merge if this PR looks good to you.

**Comment by kalefranz** [comment]
The single test failure is a one-off 502 error.

File: conda/core/solve.py [hunk 1]
```diff
@@ -397,6 +400,9 @@ def solve_for_diff(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NULL,
         # TODO: force_reinstall might not yet be fully implemented in :meth:`solve_final_state`,
         #       at least as described in the docstring.
 
+        unlink_precs = IndexedSet(reversed(sorted(unlink_precs,
```
Comment (nehaljwani) [inline conda/core/solve.py]: With this, can the following comment be removed?:
```
         # TODO: get the sort order correct for unlink_records
```

Comment (kalefranz) [reply conda/core/solve.py]: Yes, I'll remove it in that follow-on PR though that I spoke with you about.  Would rather not run CI again for this PR.

**Comment by jakirkham** [comment]
Thanks. Really excited to see this. :)

File: conda/core/solve.py (hunk 1 above)
Comment (jakirkham) [reply conda/core/solve.py]: What is the follow-on PR?

Comment (kalefranz) [reply conda/core/solve.py]: #6394

**Pull request merged**